File: src/main/java/istu/bacs/model/converter/LanguageConverter.java
package istu.bacs.model.converter;

import istu.bacs.model.type.Language;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter(autoApply = true)
public class LanguageConverter implements AttributeConverter<Language, String> {
	
	@Override
	public String convertToDatabaseColumn(Language language) {
		return language.toString();
	}
	
	@Override
	public Language convertToEntityAttribute(String languageName) {
		return Language.valueOf(languageName);
	}
}

File: src/main/java/istu/bacs/domain/LanguageConverter.kt
package istu.bacs.domain

import org.springframework.stereotype.Component
import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Component
@Converter(autoApply = true)
class LanguageConverter : AttributeConverter<Language, Int> {
    override fun convertToDatabaseColumn(language: Language?) = language!!.languageId
    override fun convertToEntityAttribute(id: Int?) = Language.findById(id!!)
}